src/test/java/teammates/test/cases/logic/EmailsTest.java

```diff
@@ -436,6 +451,79 @@ private void verifyEmail(InstructorAttributes i, MimeMessage email,
     public void testNoExceptionThrownWhenNoMessagesToSend() {
         new Emails().sendEmails(new ArrayList<MimeMessage>());
     }
+    
+    @Test
+    public void testParseMimeMessageToSendgrid() throws MessagingException, JSONException, IOException {
+        if (Config.isUsingSendgrid()) {
```

Comment:
The test for parsing MimeMessage to Sendgrid should run only if we are using Sendgrid. Otherwise, we can get an authentication error when the Sendgrid constructor is called with an username and password.